File: engine/src/main/java/ru/vertuos/engine/world/handlers/updater/PositionUpdater.java
package ru.vertuos.engine.world.handlers.updater;

import ru.vertuos.engine.world.object.DynamicGameObject;

public class PositionUpdater implements Updater<DynamicGameObject> {

    @Override
    public void update(DynamicGameObject object, float dt) {
        object.setPositionX(object.getPositionX() + object.getLinearVelocityX() * dt);
        object.setPositionY(object.getPositionY() + object.getLinearVelocityY() * dt);
    }
}

File: engine/src/main/kotlin/ru/vertuos/engine/world/updaters/updater/PositionUpdater.kt
package ru.vertuos.engine.world.updaters.updater

import ru.vertuos.engine.math.plus
import ru.vertuos.engine.math.times
import ru.vertuos.engine.world.obj.DynamicGameObject

class PositionUpdater : Updater<DynamicGameObject> {

    override fun update(obj: DynamicGameObject, dt: Float) {
        obj.position += obj.linearVelocity * dt
    }
}
